extensions/csv/src/main/java/io/deephaven/csv/parsers/context/SentinelConfiguration.java

```diff
@@ -0,0 +1,67 @@
+package io.deephaven.csv.parsers.context;
+
+/**
+ * This class holds the configuration for the optional sentinel values for each type.
+ * This is useful for systems where there is a special distinguished value taken from the
+ * range of values meant to be interpreted as null. For example, in the Deephaven system,
+ * Long.MIN_VALUE is not a valid long value in the system; rather it is taken to mean "null".
+ *
+ * When the reader encounters a null CSV cell (typically, one that is empty or has been configured
+ * by the user to be the special "null cell" string), the library will consult the values
+ * here to see if there is a sentinel value for that type that means null. If there is no
+ * such sentinel value configured, then it is an error to have a null CSV cell. Conversely,
+ * if there is a sentinel value configured here, then the system will refuse to read
+ * a value of that type as a normal value of that type.
+ *
+ * This rule interacts with type inference: say you have configured nullIntValue to be
+ * some value (say, Integer.MIN_VALUE). If the system is reading a column of ints, and
+ * you have this nullIntValue in the input, then the system will conclude that it cannot
+ * parse the column as int and it will try the next configured parser (typically, the
+ * parser for long). Each parser is configured this way (to punt to the next). If type
+ * inferencing has led you to the "last" numeric parser, namely double, and you encounter
+ * the nullDoubleValue, then the system would interpret your column as strings.
+ */
+public class SentinelConfiguration {
```

Comment:
You may be interested in using the Immutables (compile time) library to have a final POJO and builder instead of public configuration?